src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -308,8 +308,8 @@ protected void RemovePSModulePathFromEnvironment()
             }
         }
 
-        protected void PublishTelemetry<T>(
-            Dictionary<string, T> telemetryData,
+        protected void PublishTelemetry(
+            object telemetryData,
```

Comment:
added overload

Reply:
I feel like you might be missing an overload.  Otherwise, why did you change from Dictionary<string, T> to object to begin with?  I'd assumed because you needed to pass in another type.  If the change is no longer needed, better to undo...